### ISSUE TITLE
DtypeWarning on mixed types

### ISSUE THREAD

**Issue by MrAndySweet** [dgplacenames/osnb#6]
I just encountered another error message when scraping the Argyll books. I set it to work with Volumes 41 to 55 and it seems to have done that successfully, however it gave the following error message at the end of the process:

argyll-volume-55/43
argyll-volume-55/44
argyll-volume-55/45
/Users/andy/Downloads/osnb-main/volume_scraper.py:109: DtypeWarning: Columns (15,16,18,19,20,21,24,25,28,29,30,31,32) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file_path)
Finished
Exit status: 0

**Comment by MrAndySweet** [comment]
I encountered a similar error when progressing with Argyll Volumes 56 to 70 (again it seems to have completed the process).

argyll-volume-70/150
argyll-volume-70/151
argyll-volume-70/152
/Users/andy/Downloads/osnb-main/volume_scraper.py:109: DtypeWarning: Columns (16,17,22,24,25,27,28,29,30,31,32) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file_path)
Finished
Exit status: 0

**Comment by dgplacenames** [comment]
Thanks for letting me know. That's far more columns than I would have expected. I'll scrape those volumes and see what's causing it.

**Comment by MrAndySweet** [comment]
Just adding the final error message for Argyll in case it's relevant, for Volumes 71 to 81:

argyll-volume-81/151
argyll-volume-81/152
argyll-volume-81/153
/Users/andy/Downloads/osnb-main/volume_scraper.py:105: DtypeWarning: Columns (17,20,21,24,25,27,28,29,30,31,32,33,34,35,36,37) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file_path)
Finished
Exit status: 0

**Comment by dgplacenames** [comment]
Thanks again for flagging this. I ran the script on Volumes 41 to 55. I didn't get the mixed types warning but have set 'low_memory=False' all the same. The extra columns are the result of a formatting issue with certain tables on ScotlandsPlaces, e.g. https://scotlandsplaces.gov.uk/digital-volumes/ordnance-survey-name-books/argyll-os-name-books-1868-1878/argyll-volume-47/84. Not much can be done about that but it only takes a few clicks to move the columns over in Excel. Interestingly, I only got 18 columns from those volumes. Perhaps you got up to 37 because the the all_volumes csv is combing more than just the files generated from vols 41 to 55 (if you have scraped other volumes to the same folder).
